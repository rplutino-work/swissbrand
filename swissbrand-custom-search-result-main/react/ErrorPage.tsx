import React from 'react'
import { useRuntime } from 'vtex.render-runtime'

const ErrorPage: StorefrontFunctionComponent = () => {
    const runtime = useRuntime()
    const term = runtime?.query?._q

  

        if(term){
          return (
            <>
              <div className={`vtex-flex-layout-0-x-ProductErrorPage--wrapper`}>
              <h1 className={`vtex-store-components-3-x-ProductErrorPage--title`}>Su búsqueda de "{term}" no arrojó ningún resultado.</h1>
              <p className={`vtex-store-components-3-x-ProductErrorPage--subtitle`}>Compruebe los términos introducidos. Intenta utilizar una sola palabra. Utilice términos genéricos en la búsqueda. Busque utilizar sinónimos al término deseado.</p>
              <p className={`vtex-store-components-3-x-ProductErrorPage--text`}>Tambien podés visiar alguno de los productos recomendados</p>
            </div>
            </>
          )
        }

        return (
          <>
            <div className={`vtex-flex-layout-0-x-PageErrorPage--wrapper`}>
            <h1 className={`vtex-store-components-3-x-PageErrorPage--title`}>¡Perdón!, no sabemos lo que pasó.</h1>
            <p className={`vtex-store-components-3-x-PageErrorPage--text`}>La página que estás buscando no la encontramos. ¡Pero no te vayas! Te recomendamos algunos productos</p>
          </div>
          </>
        ) 
}

export default ErrorPage
