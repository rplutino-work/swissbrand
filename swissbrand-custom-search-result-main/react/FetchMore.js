import React from 'react'
import { path } from 'ramda'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import FetchMoreButton from './components/loaders/FetchMoreButton'
import LoadingSpinner from './components/loaders/LoadingSpinner'
import { PAGINATION_TYPE } from './constants/paginationType'
import { useFetchMore } from './hooks/useFetchMore'

import { useSearchPage } from 'vtex.search-page-context/SearchPageContext'

import styles from './searchResult.css'

const FetchMore = ({ htmlElementForButton = 'button '}) => {
  const { pagination, searchQuery, maxItemsPerPage, page } = useSearchPage()
  const products = path(['data', 'productSearch', 'products'], searchQuery)
  const recordsFiltered = path(
    ['data', 'productSearch', 'recordsFiltered'],
    searchQuery
  )
  const fetchMore = path(['fetchMore'], searchQuery)
  const queryData = {
    query: path(['variables', 'query'], searchQuery),
    map: path(['variables', 'map'], searchQuery),
    orderBy: path(['variables', 'orderBy'], searchQuery),
    priceRange: path(['variables', 'priceRange'], searchQuery),
  }

  const { handleFetchMoreNext, loading, to } = useFetchMore({
    page,
    recordsFiltered,
    maxItemsPerPage,
    fetchMore,
    products,
    queryData,
  })

  const isShowMore = pagination === PAGINATION_TYPE.SHOW_MORE

  if (isShowMore) {
    return (
      <div
        className={classNames(
          styles['buttonShowMore--layout'],
          'w-100 flex justify-center'
        )}
      >
        <FetchMoreButton
          products={products}
          to={to}
          recordsFiltered={recordsFiltered}
          onFetchMore={handleFetchMoreNext}
          loading={loading}
          showProductsCount={false}
          htmlElementForButton={htmlElementForButton}
        />
      </div>
    )
  }

  return <LoadingSpinner loading={loading} />
}

FetchMore.propTypes = {
  /* html element to render for fetch more button */
  htmlElementForButton: PropTypes.string,
}

export default FetchMore
