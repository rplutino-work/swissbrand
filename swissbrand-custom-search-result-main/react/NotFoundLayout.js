import React, { useMemo } from 'react'
import { SearchPageContext } from 'vtex.search-page-context/SearchPageContext'
import { generateBlockClass } from '@vtex/css-handles'

import styles from './searchResult.css'

const NotFoundLayout = ({
  children,
  searchQuery,
  maxItemsPerPage,
  map,
  params,
  priceRange,
  orderBy,
  blockClass,
}) => {
  const context = useMemo(
    () => ({
      searchQuery,
      maxItemsPerPage,
      map,
      params,
      priceRange,
      orderBy,
    }),
    [map, maxItemsPerPage, orderBy, params, priceRange, searchQuery]
  )

  return (
    <SearchPageContext.Provider value={context}>
      <div
        className={`${generateBlockClass(
          styles['notFound--layout'],
          blockClass
        )} flex flex-column flex-grow-1`}
      >
        {children}
      </div>
    </SearchPageContext.Provider>
  )
}

export default NotFoundLayout
