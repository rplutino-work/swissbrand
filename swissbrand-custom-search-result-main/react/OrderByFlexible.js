import React from 'react'
import { useSearchPage } from 'vtex.search-page-context/SearchPageContext'

import OrderBy from './OrderBy'

import styles from './searchResult.css'

const withSearchPageContextProps = Component => props => {
  const { orderBy } = useSearchPage()
  if (orderBy == null) {
    return null
  }

  return (
    <div className={styles['orderBy--layout']}>
      <Component {...props} orderBy={orderBy} />
    </div>
  )
}

const OrderByFlexible = withSearchPageContextProps(OrderBy)

OrderByFlexible.schema = {
  title: 'admin/editor.search-result.ordination.sort-by',
}

export default OrderByFlexible
