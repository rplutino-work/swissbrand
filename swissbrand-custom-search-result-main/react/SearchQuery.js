import SearchQuery from './components/SearchQuery'
export default SearchQuery
