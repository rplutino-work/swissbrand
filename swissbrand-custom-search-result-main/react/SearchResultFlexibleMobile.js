import React from 'react'
import SearchResultFlexible from './SearchResultFlexible'

const SearchResultFlexibleMobile = props => {
  return <SearchResultFlexible {...props} />
}

SearchResultFlexibleMobile.schema = {
  title: 'admin/editor.search-result-mobile.title',
}

export default SearchResultFlexibleMobile
