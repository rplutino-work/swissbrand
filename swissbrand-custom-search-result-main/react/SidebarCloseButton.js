import SidebarCloseButton from './components/SidebarCloseButton'
export default SidebarCloseButton
