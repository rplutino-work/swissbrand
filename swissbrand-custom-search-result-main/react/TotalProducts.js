import { useIntl } from 'react-intl'
import React from 'react'
import PropTypes from 'prop-types'
import { useCssHandles } from 'vtex.css-handles'
import { formatIOMessage } from 'vtex.native-types'

import styles from './searchResult.css'

const CSS_HANDLES = ['totalProductsMessage']

const TotalProducts = ({
  message = 'store/search.total-products-2',
  recordsFiltered,
  wrapperClass = styles.totalProducts,
}) => {
  const handles = useCssHandles(CSS_HANDLES)
  const intl = useIntl()
  const values = {
    recordsFiltered,
    // eslint-disable-next-line react/display-name
    span: (...chunks) => (
      <span className={`${handles.totalProductsMessage} c-muted-2`}>
        {chunks}
      </span>
    ),
  }
  const totalProductsMessage = formatIOMessage({ id: message, intl }, values)
  return (
    <div
      className={`${wrapperClass} pv5 ph9 bn-ns bt-s b--muted-5 tc-s tl t-action--small`}
    >
      <span>{totalProductsMessage}</span>
    </div>
  )
}

TotalProducts.propTypes = {
  /** Total of records filtered */
  recordsFiltered: PropTypes.number.isRequired,
  wrapperClass: PropTypes.string,
  message: PropTypes.string,
}

export default TotalProducts
