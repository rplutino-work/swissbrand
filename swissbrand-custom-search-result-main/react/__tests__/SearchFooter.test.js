/* eslint-env jest */
import { render, fireEvent } from '@vtex/test-tools/react'
import React from 'react'

import SearchFooter from '../components/SearchFooter'

describe('<SearchFooter /> component', () => {
  const getLinkProps = jest.fn().mockImplementation(pageNumber => {
    return {
      page: `mockedPage${pageNumber}`,
      params: 'mockedParams',
    }
  })
  const renderComponent = customProps => {
    const props = {
      recordsFiltered: 20,
      page: 1,
      maxItemsPerPage: 5,
      runtime: { navigate: () => {} },
      getLinkProps,
      ...customProps,
    }
    return { ...render(<SearchFooter {...props} />), getLinkProps }
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeDefined()
  })

  it('should return link', () => {
    const { getAllByTestId, getLinkProps } = renderComponent()
    const firstLink = getAllByTestId('page-button')[0]

    fireEvent.click(firstLink)

    // Expect getLinkProps has been called
    expect(getLinkProps).toBeCalledTimes(1)
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
