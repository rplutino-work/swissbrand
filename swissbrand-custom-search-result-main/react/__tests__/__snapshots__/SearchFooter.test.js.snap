// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFooter /> component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center b"
  >
    <div
      class="ph2 pointer c-on-base"
      data-testid="page-button"
    >
      1
    </div>
    <div
      class="ph2 pointer c-disabled"
      data-testid="page-button"
    >
      2
    </div>
    <div
      class="ph2 pointer c-disabled"
      data-testid="page-button"
    >
      3
    </div>
    <div
      class="ph2 pointer c-disabled"
      data-testid="page-button"
    >
      4
    </div>
    <div
      class="ph2 pointer"
    >
      <div>
        IconCaret
      </div>
    </div>
  </div>
</DocumentFragment>
`;
